src/components/TodoList/TodoList.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/label-has-for */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import TodoItem from '../TodoItem/TodoItem';
+import './TodoList.css';
+
+class TodoList extends Component {
+  state = {
+    filter: 'all',
+  }
+
+  handleClickButtonClear = () => {
+    const { items, removeTodos } = this.props;
+
+    removeTodos(items.filter(item => item.completed).map(item => item.id));
+  }
+
+  handleFilters = ({ target }) => {
+    const filter = target.getAttribute('href').slice(2) || 'all';
+
+    this.setState({
+      filter,
+    });
+  }
+
+  handleToogleAll = ({ target }) => {
+    const { checked } = target;
+    const { setCompletedAll } = this.props;
+
+    setCompletedAll(checked);
+  }
+
+  render() {
+    const {
+      items,
+      changeCompleted,
+      removeTodos,
+      editTask,
+    } = this.props;
+    const { filter } = this.state;
+
+    const createFilteredTodos = (list, switcher) => {
+      switch (switcher) {
+        case 'completed':
+          return [...list].filter(item => item.completed);
+        case 'active':
+          return [...list].filter(item => !item.completed);
+        default:
+          return [...list];
+      }
+    };
+
+    const filteredTodos = createFilteredTodos(items, filter);
+
+    return (
+      <>
+        <section className="main" style={{ display: 'block' }}>
+          <input
+            type="checkbox"
+            id="toggle-all"
+            className="toggle-all"
+            checked={filteredTodos.every(item => item.completed)}
+            onChange={this.handleToogleAll}
+          />
+          {filteredTodos.length
+            ? <label htmlFor="toggle-all">Mark all as complete</label>
+            : null
+          }
+
+          <ul className="todo-list">
+            {filteredTodos
+              .map(item => (
+                <TodoItem
+                  item={item}
+                  key={item.id}
+                  changeCompleted={changeCompleted}
+                  removeTodos={removeTodos}
+                  editTask={editTask}
+                />
+              ))}
+          </ul>
+        </section>
+
+        <footer className="footer" style={{ display: 'block' }}>
+          <span className="todo-count">
+            {`${items.filter(item => !item.completed).length} items left`}
+          </span>
+
+          <ul className="filters">
+            <li>
+              {filter === 'all'
```

Comment:
just change classnames, why you rewrite the whole element every time?

Reply:
add classnames library

Reply:
> just change classnames, why you rewrite the whole element every time?
> add classnames library

refactored